cmd/influxd/upgrade/setup.go

```diff
@@ -30,20 +30,25 @@ func onboardingRequest(ui *input.UI, options *options) (*influxdb.OnboardingRequ
 	}
 
 	req := &influxdb.OnboardingRequest{
-		User:            options.target.userName,
-		Password:        options.target.password,
-		Org:             options.target.orgName,
-		Bucket:          options.target.bucket,
-		RetentionPeriod: influxdb.InfiniteRetention,
-		Token:           options.target.token,
+		User:                   options.target.userName,
+		Password:               options.target.password,
+		Org:                    options.target.orgName,
+		Bucket:                 options.target.bucket,
+		RetentionPeriodSeconds: influxdb.InfiniteRetention,
+		Token:                  options.target.token,
 	}
 
 	if options.target.retention != "" {
 		dur, err := internal.RawDurationToTimeDuration(options.target.retention)
 		if err != nil {
 			return nil, err
 		}
-		req.RetentionPeriod = dur
+		secs := dur / time.Second
```

Comment:
As above, can you use the Duration.Seconds() call?

Reply:
Sounds like a good idea, I'll play with the math